firewood/src/storage/linear/historical.rs

```diff
@@ -34,22 +37,27 @@ impl<P: ReadLinearStore> Historical<P> {
             size,
         }
     }
-}
 
-impl<P: ReadLinearStore> ReadLinearStore for Historical<P> {
-    fn stream_from(&self, addr: u64) -> Result<Box<dyn std::io::Read + '_>, std::io::Error> {
-        Ok(Box::new(LayeredReader::new(addr, self.into())))
+    pub(super) fn stream_from(
+        &self,
+        addr: u64,
+    ) -> Result<Box<dyn std::io::Read + '_>, std::io::Error> {
+        Ok(Box::new(LayeredReader::new(
+            addr,
+            Layer::new(self.parent.clone(), &self.changed_in_parent),
+        )))
     }
 
-    fn size(&self) -> Result<u64, std::io::Error> {
+    pub(super) const fn size(&self) -> Result<u64, std::io::Error> {
```

Comment:
It doesn't really make sense to have this as a const method. When are you ever going to be creating `Historical` in a const context (at compile time)?

Or was this a clippy suggestion? I would actually ignore the rule for this if that's the case.